codalab/lib/worksheet_util.py

```diff
@@ -80,6 +83,126 @@ def worksheet_line(description, uuid):
     return '[%s]{{%s}}' % (description, uuid)
 
 
+
+############################################################
+
+
+class WorksheetResolver(object):
```

Comment:
I dramatically simplified the logic. I decided to keep canonicalize, since it's not a bad name and I was able to reduce all calls on the server side to canonicalize calls.

On the client side, I just added a few additional methods in BundleCLI that make the appropriate REST API calls to resolve uuids.